### PR TITLE
Do not call ABIEncoderV2 or the Yul optimizer experimental.

### PR DESCRIPTION
**Pull request by chriseth**
References #7970

### CODE REVIEW

File: docs/layout-of-source-files.rst [hunk 1]
```diff
@@ -70,10 +70,10 @@ ABIEncoderV2
...
 
 The new ABI encoder is able to encode and decode arbitrarily nested
-arrays and structs. It produces less optimal code (the optimizer
-for this part of the code is still under development) and has not
-received as much testing as the old encoder. You can activate it
-using ``pragma experimental ABIEncoderV2;``.
+arrays and structs. It might produce less optimal code and has not
+received as much testing as the old encoder, but is considered
+non-experimental as of Solidity 0.6.0. You still have to explicitly
+activate it using ``pragma experimental ABIEncoderV2;``.
```
Comment (bshastry) [inline docs/layout-of-source-files.rst]: This is a bit confusing. We call it non-experimental but require an `experimental` pragma to expose it.

**Review by bshastry — COMMENTED** [review]
LGTM. Documentation is a bit confusing.

File: docs/layout-of-source-files.rst (hunk 1 above)
Comment (chriseth) [reply docs/layout-of-source-files.rst]: I know :(

**Comment by leonardoalt** [comment]
Rebased

**Comment by chriseth** [comment]
@bshastry tried to improve the wording a little, please see if this is better.

**Review by leonardoalt — APPROVED** [review]

**Review by bshastry — APPROVED** [review]
Looks good to me.

**Pull request merged**